server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java

```diff
@@ -789,6 +799,26 @@ public Builder dropRelation(RelationName relationName) {
             return this;
         }
 
+        @Nullable
+        @SuppressWarnings("unchecked")
+        public <T extends RelationMetadata> T getRelation(RelationName relation) {
```

Comment:
Could factor out either a `private static <T extends RelationMetadata> T getRelation(SchemaMetadata schemaMetadata, RelationName relation) {` or one with a `schemas::get` `Function<String, SchemaMetadata>` parameter and reuse it for both this and the `Metadata.getRelation`

Reply:
👍 thx, done.